core/src/main/java/com/themecleanflex/models/ReferenceModel.java

```diff
@@ -605,18 +610,66 @@ private String generateReferenceJson() {
       return null;
     }
 
-    // find a node with the given key/value pair in our json output
+	private boolean createsReferenceLoop(final Resource resource, final Set<String> forbiddenPaths) {
```

Comment:
instead of having to pass `new HashSet<>()` wouldn't it make more sense to set that as default-value?

Reply:
We could add another method for that, but it's subsequently meant to be called recursively. The first call requires an empty map. We'd not benefit much from the new proxy method in my opinion though.